pkg/apis/core/validation/validation.go

```diff
@@ -3972,17 +3974,19 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 
 	// handle updateable fields by munging those fields prior to deep equal comparison.
 	mungedPod := *newPod
-	// munge spec.containers[*].image
+	// munge spec.containers[*].image and imagePullPolicy for AlwaysPullImages admission controller
 	var newContainers []core.Container
 	for ix, container := range mungedPod.Spec.Containers {
 		container.Image = oldPod.Spec.Containers[ix].Image
+		container.ImagePullPolicy = oldPod.Spec.Containers[ix].ImagePullPolicy
```

Comment:
Allow changing ImagePullPolicy will cause kubelet to restart container. Is it reasonable?

Reply:
I changed the solution to `ignore update pod in alwayspullimages admission controller` now. 
Thanks for your comments.

Reply:
I guess removing Update from operation would do the same and is easier:
https://github.com/kubernetes/kubernetes/blob/e51d3c92398b31d405483d66b0ebd7b33c5271a2/plugin/pkg/admission/alwayspullimages/admission.go#L120

Reply:
> I guess removing Update from operation would do the same and is easier:

we should only ignore Update if no images were changed in the new pod spec

Reply:
Didn't expect your quick response to my deleting comment. LOL. I agree. Taking the image change into the consideration is a better approach.